### ISSUE TITLE
Rule Proposal: Warn if "by" objects are passed to $ and $$ 

### ISSUE THREAD

**Issue by alecxe** [alecxe/eslint-plugin-protractor#66]
Inspired by http://stackoverflow.com/q/41598550/771848. 

Invalid usage:

```javascript
$$(by.repeater('post in posts'));
$(by.css('[ng-click="posts.newPost()"]'))
```

Correct usage:

```javascript
element.all(by.repeater('post in posts'));
$('[ng-click="posts.newPost()"]');
```

May be we should simply check if the argument to $ or $$ is a string. To be discussed.

**Comment by alecxe** [comment]
Already covered by [`valid-locator-type`](https://github.com/alecxe/eslint-plugin-protractor/blob/master/docs/rules/valid-locator-type.md).
